charset/charset.go

```diff
@@ -421,6 +431,7 @@ var collations = []*Collation{
 	{246, "utf8mb4", "utf8mb4_unicode_520_ci", false},
 	{247, "utf8mb4", "utf8mb4_vietnamese_ci", false},
 	{255, "utf8mb4", "utf8mb4_0900_ai_ci", false},
+	{2048, "utf8mb4", "utf8mb4_zh_tidb_pinyin_cs", false},
```

Comment:
```suggestion
	{2048, "utf8mb4", "utf8mb4_tidb_zh_pinyin_cs", false},
```
How about this?

Reply:
Please update the PR description accordingly.

Reply:
the naming convention in mysql,  `zh` follows `utf8mb4` and `cs` at last, `tidb_pinyin` can be seems as a special version.

Reply:
tidb is the version, and `pinyin` is the variant. this is similar to `utf8mb4_de_pb_0900_ai_ci` (pb = phonebook) and `utf8mb4_es_trad_0900_ai_ci` (trad = traditional).

Reply:
updated name to `utf8mb4_zh_pinyin_tidb_as_cs`